samples/nrf9160/http_application_update/src/main.c

```diff
@@ -131,19 +126,18 @@ static int flash_page_erase_if_needed(u32_t address)
 }
 
 
-static int app_download_client_event_handler(
-	struct download_client *const dfu,
-	enum download_client_evt event, u32_t error)
+static int download_client_callback(const struct download_client_evt *event)
 {
 	int err;
 
-	switch (event) {
-	case DOWNLOAD_CLIENT_EVT_DOWNLOAD_FRAG: {
-
-		if (dfu->object_size > PM_MCUBOOT_SECONDARY_SIZE) {
+	switch (event->id) {
+	case DOWNLOAD_CLIENT_EVT_FRAGMENT: {
+#if 0
+		if (dfu->file_size > PM_MCUBOOT_SECONDARY_SIZE) {
```

Comment:
I think the best way to return the file size would be via an event, and the simplest way could be to add it to an existing event instead of sending out a separate one, just to keep the state machine simple.

Reply:
Added a dedicated API.